### DISCUSSION TITLE
请问能否增加以管理员权限开机启动的功能，比较习惯于管理员模式使用软件

### DISCUSSION THREAD

**Discussion by Radjedef-Iris** [shuax/MouseInc.Settings#317] [Ideas]
管理员模式使用软件可以对所有软件进行手势操作，比较方便

**Comment by 6Tom** [comment]
我做了两个万能批处理，只要是目录下扩展名为exe的只有一个的软件，添加开机以管理员自启都能用；如果有多个exe的，就自己定制一下，把“*.exe”改成想要的名字

放在mouseinc同一目录下，双击设置/清除任务计划

设置任务计划.bat
```
cd /d "%~dp0"
set "file_=*.exe"
if not exist "%file_%" goto :end
%1 start "" mshta vbscript:createobject("shell.application").shellexecute("""%~0""","::",,"runas",1)(window.close) & exit
for /f "delims=" %%a in ('dir /b /a-d /on "%file_%"') do set "name_=%%~na" & set "path_=%%~dpnxa"
schtasks /create /tn "%name_%" /tr "\"%path_%\"" /sc onlogon /rl highest /f
:end
```

清除任务计划.bat
```
cd /d "%~dp0"
set "file_=*.exe"
if not exist "%file_%" goto :end
%1 start "" mshta vbscript:createobject("shell.application").shellexecute("""%~0""","::",,"runas",1)(window.close) & exit
for /f "delims=" %%a in ('dir /b /a-d /on "%file_%"') do set "name_=%%~na"
schtasks /delete /tn "%name_%" /f
:end
```

**Comment by Bush2021** [comment]
![MouseInc_以管理员权限运行](https://user-images.githubusercontent.com/79072750/181291583-5a7c5315-1f69-4a7b-ab64-1fff61e8d455.jpg)
请先查看使用手册和软件设置再提问

**Reply by 6Tom** [reply]
不一样的兄弟，一看你就是关闭UAC用户账户控制弹窗的人，所以你只要勾上图示这个，再勾上自启就完事了

你把UAC弹窗打开试试就明白了，开机还要弹窗问你确认一遍管理员，请求升权是很好的安全策略，很多人都特意开着，还有一批人是因为不懂所以没概念要关

正确的做法应该是添加任务计划，让mouseinc在用户登录时以最高权限运行，并且你才是应该看手册的人，手册里写了任务计划的事，只不过没有提供一键设置的途径，所以题主是想求这个事来了，这个提问是完全合理的

**Reply by Bush2021** [reply]
呃, 手册的截图忘记贴上来了……
![2022-07-28_123258](https://user-images.githubusercontent.com/79072750/181421169-dfbd76d2-cbf3-4adc-96e6-526d21ca8e16.jpg)

题主并没有明确自己是否开启 UAC 提权, 所以我希望他查看手册和软件设置自行解决此问题
无论哪种情况手册都能解决, 并且操作方式写的很清楚了, 还不看手册就提问, 合理吗?
https://docs.shuax.com/MouseInc/#/faq#%E5%A6%82%E4%BD%95%E8%AE%BE%E7%BD%AE%E4%BB%BB%E5%8A%A1%E8%AE%A1%E5%88%92%E5%90%AF%E5%8A%A8%EF%BC%9F

另外我不讨论请求提权是否明智, 但请不要以为不开 UAC 的都是对此没有概念的人 (我自己是真觉得累赘, 能混 Github 的人对软件、安全还是略知一二的

**Reply by 6Tom** [reply]
大家都是好心帮忙，不用吵架，重点是帮耍下大佬解决下力所能及的问题，省着大佬事太多，关注解决问题即可

我是嫌动手设这设那很烦那种人，所以能一键的事，只要能做出来，我是说啥都会先做出来，然后一劳永逸

就如我说手册上是特意写了，但是没有一键的，所以这个提问是合理的，无非就是在跟大佬表达，希望可以把一键任务计划集成进mouseinc里呗

既然现在没集成，那就换个也很简单的方式，外部集成，亦是轻松+愉快
